hieradata/vagrant/common.yaml

```
@@ -54,6 +54,10 @@ profile::jenkinscontroller::jcasc:
         # but hierdata is behaving weirdly (bug in YAML parser?)
         defaultDisplayUrlProvider:
           providerId: "org.jenkinsci.plugins.displayurlapi.ClassicDisplayURLProvider"
+  csp-plugin:
+    reportOnly: true
+    rule: |-
+      "default-src 'self'; style-src 'self' 'unsafe-inline'; img-src 'self' data: avatars.githubusercontent.com; media-src 'self' data: ; script-src 'self' 'report-sample' usage.jenkins.io;"
```

I don't think `report-sample` is a thing, I think @daniel-beck made a typo in his comment.

Shouldn't it be this?
```suggestion
      "default-src 'self'; style-src 'self' 'unsafe-inline'; img-src 'self' data: avatars.githubusercontent.com; media-src 'self' data: ; script-src 'self'; report-to usage.jenkins.io;"
```

To support firefox you may also want this:

```suggestion
      "default-src 'self'; style-src 'self' 'unsafe-inline'; img-src 'self' data: avatars.githubusercontent.com; media-src 'self' data: ; script-src 'self'; report-to usage.jenkins.io; report-uri https://usage.jenkins.io"
```

---

cc @daniel-beck

---

https://github.com/jenkins-infra/helpdesk/issues/4776#issuecomment-3206262731 has the exact string that should be used.

The `report-uri` directive is dynamically generated, so while you are correct in isolation/in general, it doesn't apply to CSP plugin. (Also, `usage.jenkins.io` does not _accept_ reports, it _hosts a script_ for usage stats we allow with this rule. So it's an additional allowed source.)

> I don't think `report-sample` is a thing

It is, and gives a snippet of the script. See e.g. https://github.com/jenkinsci/stapler/pull/612 for an attempt at adding the source file path to it.

---

Ah I see it here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Content-Security-Policy#report-sample doesn't come up on google very well